src/main/java/org/spongepowered/api/item/inventory/Inventory.java

```diff
@@ -505,6 +505,33 @@
     InventoryArchetype getArchetype();
 
     /**
+     * Intersects the slots of both inventories
+     * The resulting inventory will only contain slots.
+     *
+     * @param inventory the other inventory
+     * @return an inventory wrapping all slots that are present in both inventories
+     */
+    Inventory intersect(Inventory inventory);
+
+    /**
+     * Constructs a union of the slots in both inventories.
+     * The resulting inventory will only contain slots.
+     *
+     * @param inventory the other inventory
+     * @return an inventory wrapping all slots of both inventories.
+     */
+    Inventory union(Inventory inventory);
+
+    /**
+     * Returns true if the given inventory is a descendant of this one.
+     * This includes deeply nested ones but not usually not query-results.
+     *
+     * @param inventory the other inventory
+     * @return whether the given inventory is contained in this one.
+     */
+    boolean containsInventory(Inventory inventory);
```

Comment:
I would rephrase or extend the javadoc by something like:
* `an inventory is contained in another if all slots are contained in the parent`
* `an inventory is a decendant of an inventory if all slots of the decendant are contained in the parent`

This method can also be default implemented using:
`this.union(inventory).capacity() == this.capacity()`